### ISSUE TITLE
Upgrade from 4.15.z to 4.17 d/s builds, rsyslog env has an issue, noobaa.log is not updated

### ISSUE THREAD

**Issue by rkomandu** [noobaa/noobaa-core#8336]
### Environment info
- NooBaa Version: **VERSION**
- Platform: **Kubernetes 1.14.1 | minikube 1.1.1 | OpenShift 4.1 | other: specify**

Have a CES S3 cluster with noobaa 4.15.6 version 

### Actual behavior
1. 

Upgraded on the nodes RH 8.x and 9.x as shown below 

uname -a 
Linux rk521-11 4.18.0-513.5.1.el8_9.x86_64 #1 SMP Fri Sep 29 05:21:10 EDT 2023 x86_64 x86_64 x86_64 GNU/Linux
Linux rk521-12 5.14.0-427.13.1.el9_4.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Apr 10 10:29:16 EDT 2024 x86_64 x86_64 x86_64 GNU/

rk521-11) rpm -qi noobaa-core-5.17.0-20240904.el8.x86_64 | grep Install
Install Date: Mon 09 Sep 2024 10:52:16 AM CEST

current time -->  date
Mon Sep  9 11:26:21 CEST 2024

system.json file  (snippet) 
rk521-11":{"current_version":"5.17.0","upgrade_history":{"successful_upgrades":[{"timestamp":1725872074467,"completed_scripts":[],"from_version":"5.15.4","to_version":"5.17.0"}]}}
rk521-12":{"current_version":"5.17.0","upgrade_history":{"successful_upgrades":[{"timestamp":1725872767473,"completed_scripts":[],"from_version":"5.15.4","to_version":"5.17.0"}]}}}

In the noobaa file present in the logrotate.d directory (boobaa-logrotate) didn't have any further updates into noobaa.log file 

[rk521-12 log]# rpm -qa |grep noobaa
noobaa-core-5.17.0-20240904.el9.x86_64
[rk521-12 log]# rpm -qi noobaa-core-5.17.0-20240904.el9.x86_64 | grep -i Install
Install Date: Mon 09 Sep 2024 11:05:13 AM CEST

rpm -qf /etc/logrotate.d/noobaa-logrotate
noobaa-core-5.17.0-20240904.el8.x86_64

/etc/logrotate.d/noobaa-logrotate  (in the 5.17.0-xxx) below snippet  
/var/log/noobaa.log
{
    daily
    maxsize 100M
    minsize 50M
    start 1
    missingok
    rotate 100
    compress
    create 644 root root
    sharedscripts
    postrotate
        killall -HUP rsyslogd || true
        killall -HUP syslogd || true
        systemctl reload syslog-ng > /dev/null 2>&1 || true
    endscript
}

In the 5.15.4

snippet of logrotate_noobaa.conf 

/var/log/noobaa.log
{
    daily
    maxsize 100M
    minsize 50M
    start 1
    missingok
    rotate 100
    compress
    create 644 root root
    sharedscripts
    postrotate
        killall -HUP rsyslogd || true
        killall -HUP syslogd || true
    endscript
}

### Expected behavior
1.

It should update the noobaa.log files with the Upgrade / messages as well, however it only show them in the /var/log/messages 

```
Sep  9 11:06:07 rk521-12 node[1647542]: 2024-09-09 11:06:07.356026 [PID-1647542/TID-1647542] FS::GPFS found GPFS lib file GPFS_DL_PATH=/usr/lpp/mmfs/lib/libgp
fs.so
Sep  9 11:06:07 rk521-12 node[1647542]: 2024-09-09 11:06:07.361138 [PID-1647542/TID-1647542] FS::set_debug_level 0
Sep  9 11:06:07 rk521-12 node[1647542]: 2024-09-09 11:06:07.361303 [PID-1647542/TID-1647542] FS::set_log_config: LOG_TO_STDERR_ENABLED=1 LOG_TO_SYSLOG_ENABLED
=0
Sep  9 11:06:07 rk521-12 node[1647542]: ESC[32mSep-9 11:06:07.445ESC[35m [Upgrade/1647542] ESC[36m   [L0]ESC[39m UPGRADE:: upgrade manager started..
Sep  9 11:06:07 rk521-12 node[1647542]: ESC[32mSep-9 11:06:07.447ESC[35m [Upgrade/1647542] ESC[36m  [LOG]ESC[39m CONSOLE:: read_rand_seed: reading 32 bytes fr
om /dev/urandom ...
Sep  9 11:06:07 rk521-12 node[1647542]: ESC[32mSep-9 11:06:07.463ESC[35m [Upgrade/1647542] ESC[36m  [LOG]ESC[39m CONSOLE:: read_rand_seed: got 32 bytes from /
dev/urandom, total 32 ...
Sep  9 11:06:07 rk521-12 node[1647542]: ESC[32mSep-9 11:06:07.463ESC[35m [Upgrade/1647542] ESC[36m  [LOG]ESC[39m CONSOLE:: read_rand_seed: closing fd ...
Sep  9 11:06:07 rk521-12 node[1647542]: ESC[32mSep-9 11:06:07.468ESC[35m [Upgrade/1647542] ESC[36m  [LOG]ESC[39m CONSOLE:: init_rand_seed: seeding with 32 byt
es
Sep  9 11:06:07 rk521-12 node[1647542]: ESC[32mSep-9 11:06:07.472ESC[35m [Upgrade/1647542] ESC[36m   [L0]ESC[39m UPGRADE:: container version is 5.17.0 and ser
ver version is 5.15.4. will upgrade
Sep  9 11:06:07 rk521-12 node[1647542]: ESC[32mSep-9 11:06:07.475ESC[35m [Upgrade/1647542] ESC[33m [WARN]ESC[39m UPGRADE:: upgrade scripts directory "/usr/loc
al/noobaa-core/src/upgrade/nsfs_upgrade_scripts" was not found. treating it as empty
Sep  9 11:06:07 rk521-12 node[1647542]: ESC[32mSep-9 11:06:07.476ESC[35m [Upgrade/1647542] ESC[36m   [L0]ESC[39m UPGRADE:: found the following versions with u
pgrade scripts which are newer than server version (5.15.4): []
Sep  9 11:06:07 rk521-12 node[1647542]: ESC[32mSep-9 11:06:07.603ESC[35m [Upgrade/1647542] ESC[36m   [L0]ESC[39m UPGRADE:: upgrade completed successfully!
Sep  9 11:06:07 rk521-12 systemd[1]: Started The NooBaa service..

```
For all noobaa related logs, it should be in the noobaa.log as well, Have their been changes in this area in 5.17 ?  

### Steps to reproduce
1.

Install the noobaa rpm of 4.15.4 and Upgrade to noobaa rpm to 4.17 
noobaa-core-5.15.4-20240704.el8.x86_64 --> noobaa-core-5.17.0-20240904.el8.x86_64)

**Comment by rkomandu** [comment]
It has been discussed further over the slack channel, that this is implemented in this way, which means the 4.17 noobaa rpm onwards the /var/log/messages would get the noobaa messages logged instead of noobaa.log file. 

There are couple of env variables which exists, that can be tried to change the logging behavior. However the default log into messages file of /var/log is a concern IMO, as all debug log levels also will be sent to a system wide log file which shouldn't be the case

**Comment by rkomandu** [comment]
As per Navin Paul and Romy, the final decision for logging in 5.17 is to get default logged into /var/log/messages. If we change the config parameters then it would get to both the files (i.e messages, noobaa.log) . 

Hence closing this. 
Ramya has opened a separate defect w/r/t the STDERR and SYSLOG enabled printing in the console.
